### PR TITLE
Clarify `polar.from_*` clones data

### LINKED ISSUE #4523: Unclear documentation for `polar.from_*`
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of polars.

### Issue Description

I am reporting this as a documentation bug. It is not clear whether the `polar.from_*` operations are zero-copy or not. On the contrary, both `polar.DataFrame.to_pandas` and `polar.DataFrame.to_numpy` clearly states "This operation clones data", which I find very helpful.

### Reproducible Example

```python
https://pola-rs.github.io/polars/py-polars/html/reference/api/polars.from_pandas.html
https://pola-rs.github.io/polars/py-polars/html/reference/api/polars.from_numpy.html
```

### Expected Behavior

The API document should specify whether the `polar.from_*` operations copy data from NumPy or Pandas.

### Installed Versions

<details>

```
---Version info---
Polars: 0.14.2
Index type: UInt32
Platform: macOS-12.5.1-x86_64-i386-64bit
Python: 3.8.9 (default, May 17 2022, 12:55:41)
[Clang 13.1.6 (clang-1316.0.21.2.5)]
---Optional dependencies---
pyarrow: <not installed>
pandas: 1.4.3
numpy: 1.23.2
fsspec: <not installed>
connectorx: <not installed>
xlsx2csv: <not installed>
```
</details>

### PR DESCRIPTION
**Pull request by Chuxiaof**
Fix #4523.

### CODE REVIEW

File: py-polars/polars/convert.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def from_dict(
     columns: Sequence[str] | None = None,
 ) -> DataFrame:
     """
-    Construct a DataFrame from a dictionary of sequences.
+    Construct a DataFrame from a dictionary of sequences. This operation clones data.
```
Comment (ritchie46) [inline py-polars/polars/convert.py:37]: In this case it may clone data. You can also pass a `dict[str, pl.Series]`. This does not clone data.

Comment (Chuxiaof) [reply py-polars/polars/convert.py:37]: I see. I guess this also applies to `pl.from_dicts`?

File: py-polars/polars/convert.py [hunk 2]
```diff
@@ -73,7 +74,8 @@ def from_dicts(
     dicts: Sequence[dict[str, Any]], infer_schema_length: int | None = 50
 ) -> DataFrame:
     """
-    Construct a DataFrame from a sequence of dictionaries.
+    Construct a DataFrame from a sequence of dictionaries. This operation clones data,
+    unless you pass in a `dict[str, pl.Series]`.
```
Comment (ritchie46) [inline py-polars/polars/convert.py:78]: This is not correct. `from_dicts` is row oriented and will therefore always clone data to column orientation.

Comment (Chuxiaof) [reply py-polars/polars/convert.py]: OK. Fixed that!

**Comment by ritchie46** [comment]
Thanks @Chuxiaof!

**Pull request merged**